pkg/customresourcestate/generate/generator/testdata/foo_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2023 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Changes to this file may require to regenerate the `foo-config.yaml`. Otherwise the
+// tests in ../generate_integration_test.go may fail.
+// The below marker can be used to regenerate the `foo-config.yaml` file by running
+// the following command:
+// $ go generate ./pkg/customresourcestate/generate/generator/testdata
+//go:generate sh -c "go run ../../../../../ generate ./... > foo-config.yaml"
+
+// +groupName=bar.example.com
+package foo
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FooSpec is the spec of Foo.
+type FooSpec struct {
+	// This tests that defaulted fields are stripped for v1beta1,
+	// but not for v1
+	DefaultedString string `json:"defaultedString"`
+}
+
+// FooStatus is the status of Foo.
+type FooStatus struct {
+	// +Metrics:stateset:name="status_condition",help="The condition of a foo.",labelName="status",JSONPath=".status",list={"True","False","Unknown"},labelsFromPath={"type":".type"}
+	// +Metrics:gauge:name="status_condition_last_transition_time",help="The condition last transition time of a foo.",valueFrom=.lastTransitionTime,labelsFromPath={"type":".type","status":".status"}
+	Conditions Condition `json:"conditions,omitempty"`
+}
+
+// Foo is a test object.
+// +Metrics:gvk:namePrefix="foo"
+// +Metrics:labelFromPath:name="name",JSONPath=".metadata.name"
+// +Metrics:gauge:name="created",JSONPath=".metadata.creationTimestamp",help="Unix creation timestamp."
+// +Metrics:info:name="owner",JSONPath=".metadata.ownerReferences",help="Owner references.",labelsFromPath={owner_is_controller:".controller",owner_kind:".kind",owner_name:".name",owner_uid:".uid"}
```

Comment:
Forget to review API.

QQ: what is `JSONPath=".metadata.ownerReferences"`?  Is it the value of a metric?